### ISSUE TITLE
Refining the permalink structure

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/gutenberg#12714]
**Describe the bug**
I am looking at the Permalink flow. There are a few things that I notice.

**To Reproduce**
1. Create a new post/page. Add a title. No permalink structure is seen for a few seconds.

![screen shot 2018-12-09 at 12 48 44](https://user-images.githubusercontent.com/5323259/49696880-ed5e5b80-fbb0-11e8-93a5-9d5133cc2a83.png)

Suddenly the permalink structure shows up. I believe it is the autosave that kicks in.
Permalink is listed by ?p=410 the id and not the title. Next to it is a button that says Change Permalinks. Clicking it leads to Settings -> Permalinks. The same is reflected in the Document settings Permalink area. It shows ID even though Permalinks is set to show post name.

![screen shot 2018-12-09 at 12 50 40](https://user-images.githubusercontent.com/5323259/49696902-1d0d6380-fbb1-11e8-82fd-ac684d1092af.png)

After clicking Save Draft the permalink changes to show the title.

![screen shot 2018-12-09 at 12 53 39](https://user-images.githubusercontent.com/5323259/49696952-c3f1ff80-fbb1-11e8-987b-643adf9a1ca5.png)

Bottom line is:
Upon creating a new page or post. Then filling in the title autosave seems to be kicking in but the permalink url and button next to it does not change before the user clicks Save Draft. It can be confusing in those few moments before user clicks the Save Draft button.

**Desktop (please complete the following information):**
 - OS: Mac OS X 10.13.6
 - Browser Desktop Server and Chrome
 - Version WordPress 5.0

I believe the above issue is somewhat a copy of: https://github.com/WordPress/gutenberg/issues/7129

Another: https://github.com/WordPress/gutenberg/issues/1285

**Comment by bupotalovo** [comment]
- Related: https://github.com/WordPress/gutenberg/issues/12498
- General discussion "global site settings should not be in content editor" at https://github.com/WordPress/gutenberg/issues/11945 and https://github.com/WordPress/gutenberg/issues/11970

**Comment by Wolfneo** [comment]
Hi,  
Is it normal that if I put a comma or other punctuation, I find him in the permalink in the slug?
Until WordPress opts for Gutenberg, I couldn't have this problem.
Is this a bug or is this normal? 
Thank you

**Comment by pawel-tomkiel** [comment]
@Wolfneo - even # will be inserted into permalink as I've seen. But that's probably offtopic in this issue.

**Comment by Wolfneo** [comment]
Hi,
I  do not speak English, I use a translator to put words, I just wanted to report this problem.

**Comment by greatislander** [comment]
@Wolfneo That issue has been reported elsewhere:

- https://core.trac.wordpress.org/ticket/45588
- https://github.com/WordPress/gutenberg/issues/12907

**Comment by Wolfneo** [comment]
Thank you I deleted my messages, sorry

**Comment by SenorFusion** [comment]
I feel this should _really_ be a **High Priority issue.**

Allowing posts to be published with these terrible URLs even though permalink settings are explicitly the opposite is really bad.

There must be ten of thousands of regular bloggers who don't use the "Save Draft" function and are unknowingly publishing their posts with these terrible permalinks.

Sure, the permalink can be updated - BUT once a post goes live and is pushed out to Pingomatic (and search engines), RSS, Email Notifications, Social Media, etc it is a significant problem for the success of the post (marketing, sharing, and seo wise) to change the URL after the fact.

Several non-technical bloggers that I support have had _exactly_ this problem for _exactly_ this reason. I am having to try to retrain them to manually edit the permalink (they shouldn't have to).

As mentioned above, _especially_ because of Gutenberg presenting itself as "just write".

This is a significant regression and should be moved up the priority list.

**Comment by youknowriad** [comment]
closed by #12009
